### DISCUSSION TITLE
phone number

### DISCUSSION THREAD

**Discussion by fasatrix** [faker-js/faker#2577] [Q&A]
Hi,
I was pretty disappointed to find out that the `faker.phone(<format>)` was deprecated. I tried to use the `faker.helpers.fromRegExp() `method but it generates (at least for me) an odd string as an output. How would one create a phone number let's say beginning with +39 as a country code and followed by some random digits?

**Comment by ST-DDT** [comment]
FFR: https://fakerjs.dev/api/phone.html#number

The format parameter and the regex method have similar capabilities but slightly different syntax. 

As for why:
- Because we plan to improve the method to generate the format that you need directly without having to pass a format. See also: https://github.com/faker-js/faker/issues/1542

As for the alternative (not tested):
- `faker.helpers.fromRegExp(/\+39[0-9]{10}/)`
